client/src/components/ScheduleGrid.tsx

```diff
@@ -0,0 +1,157 @@
+import React from "react";
+import { SectionWrapper } from "./Home";
+import clsx from "clsx";
+import { DataGrid, CellClassParams, isArray } from "@material-ui/data-grid";
+import { makeStyles } from "@material-ui/core/styles";
+
+
+const weekdayWidth = 125;
+const timeWidth = 90;
+const titleHeight = 40;
+const rowHeight = 30;
+
+interface ColDef {
+    /**
+     * The column identifier. It's used to match with [[RowData]] values.
+     */
+    field: string;
+    headerName: string;
+    type: string;
+    width: number;
+    sortable: boolean;
+    headerAlign: "center";
+    cellClassName: any;
+}
+
+let color = (timeslot: CellClassParams) => clsx("selected", { true: (timeslot.value as string) !== "", });
+var columns: ColDef[] = [
+    {field: "id", headerName: "Time", type: "string", width: timeWidth, sortable: false, headerAlign: "center", cellClassName: null},
+    {field: "Monday", headerName: "Mon", type: "string", width: weekdayWidth, sortable: false, headerAlign: "center", cellClassName: color, },
+    {field: "Tuesday", headerName: "Tues", type: "string", width: weekdayWidth, sortable: false, headerAlign: "center", cellClassName: color, },
+    {field: "Wednesday", headerName: "Wed", type: "string", width: weekdayWidth, sortable: false, headerAlign: "center", cellClassName: color, },
+    {field: "Thursday", headerName: "Thur", type: "string", width: weekdayWidth, sortable: false, headerAlign: "center", cellClassName: color, },
+    {field: "Friday", headerName: "Fri", type: "string", width: weekdayWidth, sortable: false, headerAlign: "center", cellClassName: color, },
+];
+
+
+let makerows = function (start: number, end: number) {
+    var rows: any[] = [];
+    let startHour = start, endHour = end, half = 0;
+    for (var hour = startHour; hour <= endHour; hour += 0.5) {
+        let minutes = (half % 2) === 1 ? ":30" : ":00";
+        var time = String(Math.floor(hour)) + minutes;
+        rows.push({
+            id: time,
+            Monday: "",
+            Tuesday: "",
+            Wednesday: "",
+            Thursday: "",
+            Friday: "",
+        }, );
+        half++;
+    }
+    return rows;
+};
+
+
+function addCourse(rows: any, day: any, startTime: number, endTime: number, courseName: string) {
+    if (startTime >= endTime || courseName === "" || courseName === null) return;
+    let colnum = day + 1;
```

Comment:
This variable is currently greyed out, but everything looks functional. Is `colnum` necessary anywhere?